src/averageAges.js

```diff
@@ -37,9 +45,23 @@
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
+  }
```

Comment:
Your implementation correctly considers the 'withChildren' condition to filter women. However, the same issue as in the men's average age calculation exists here. You're filtering based on the subset 'womens', which is not correct. You need to filter based on the entire 'people' dataset.